instance/models/openedx_appserver.py

```diff
@@ -278,9 +278,12 @@ def make_active(self, active=True):
         self.is_active = active
         self.save()
         self.instance.reconfigure_load_balancer()
+        appserver_num = self.name.split()[-1]
```

Comment:
@pkulkark, I am not sure you understood the logic here right. The number to be used in the `vmX.<instance domain>` DNS record has nothing to do with the appserver number shown in the name. It is based on the number of active appservers - if there N active appservers, DNS records `vm<1-N>.<instance domain>` will be set for the active appservers from oldest to the newest. So for example, if the appservers 11, 12 are the active ones, the `vm1` record will point to the IP address of the appserver 11 and `vm2` record will point to the IP address of the appserver 12. Then when 2 new appservers 13, 14 are spawned to replaces the existing pair and are activated, they will initially get assigned `vm4`, `vm5` and then once the appservers 11, 12 are deactivated, the `vm1`, `vm2` records will get assigned to the appservers 13, 14 and the `vm3`, `vm4` records will get deleted.

Reply:
@lgp171188 Based on your [previous comment](https://github.com/open-craft/opencraft/pull/521#discussion_r376969949), I was under the impression that the logic for `set_active_vms_dns_records` is flawed and needs fixing. Sorry if I misunderstood. 
This was one approach I thought that might make it simpler - i.e. use the appserver number for the dns record and delete them when deactivating. 
I think what you're suggesting is to keep the same logic for setting the dns record, and add a new step to it to delete the non-used dns records i.e. if appserver 11 and 12 are active initially, `vm1` and `vm2` dns records will be associated with their corresponding ip address, and when we spawn and activate a new appserver 13, then that gets `vm3` dns record. Now if we deactivate appserver 11, then `vm1` and `vm2` get assigned to appserver 12 and 13, and the new step needs to delete `vm3`. Is that correct? If so, my question is how do we get the no-longer-used dns records? do we just iterate through the remaining appservers and delete `vm<i>` dns records, where, number_of_active_appservers < i < total_number_of_spawned_appservers?
Would that not add unnecessary overhead if we have a lot of inactive appservers and this would get called everytime we activate an appserver?

Reply:
@pkulkark, see [this comment](https://github.com/open-craft/opencraft/pull/521/files#r380140691) of mine. I meant deleting the `vm<N+1>.<instance domain>` where N is the number of active appservers. The DNS client API will not throw an error when deleting a non-existent record. So if the record exists, it gets deleted. If it doesn't, nothing happens.

This will require some testing in the case of instances which should always have at least 2 active appservers, the reason being, once the 2 new appservers are active, it is very common to deactivate both older appservers in quick succession. So if the 2nd deactivation starts running before the first one finishes the reassignment, there will be an undeleted, dangling record `vm3.<instance domain>` left after the process.